microcosm_flask/tests/conventions/test_command.py

```diff
@@ -0,0 +1,131 @@
+"""
+Command operation tests.
+
+"""
+from json import dumps, loads
+from hamcrest import (
+    assert_that,
+    equal_to,
+    is_,
+)
+
+from marshmallow import fields, Schema
+from microcosm.api import create_object_graph
+
+from microcosm_flask.conventions.encoding import dump_response_data, load_request_data
+from microcosm_flask.conventions.registry import request, response
+from microcosm_flask.conventions.swagger import register_swagger_endpoint
+from microcosm_flask.operations import Operation
+
+
+class CommandArgumentSchema(Schema):
+    value = fields.String(required=True)
+
+
+class CommandResultSchema(Schema):
+    result = fields.Boolean(required=True)
+    value = fields.String(required=True)
+
+
+def make_command(graph, request_schema, response_schema):
+    """
+    Create an example command route.
+
+    """
+
+    @graph.route("/v1/foo/do", Operation.Command, "foo")
```

Comment:
This is how a command might look:
 - the URI is user-defined, but need to have the right prefix if you want swagger to pick it up
 - the final "foo" argument is essentially a namespace; it should be different for every command
 - the `@request` and `@response` decorators tell the swagger layer what schemas are in use
 - the doc string gets picked up in the swagger definition
 - the function body uses boilerplate to decode the request and encode the response
 - a "real" command would do something interesting to generate the `response_data` from the `request_data`; this logic can be extended to use path variables; just look at the use of `merge_data()` in `conventions/crud.py`.